torchvision/prototype/datasets/decoder.py

```diff
@@ -0,0 +1,40 @@
+import collections.abc
+import io
+from typing import Callable
+
+import numpy as np
+import PIL.Image
+import torch
+from torch.utils.data import IterDataPipe, functional_datapipe
+
+__all__ = ["pil"]
+
+
+@functional_datapipe("decode_sample")
```

Comment:
I've mentioned this to @VitalyFedyunin in the past, but I think that the fluent API should be used with care.

The reason is that things like
```python
import torch.utils.data
import torchvision.prototype

generic_datapipe.decode_sample(...)
```
works but
```python
import torch.utils.data

generic_datapipe.decode_sample(...)
```
doesn't can be very surprising to users, and it is very very hard to spot what is wrong / missing.

IMO I'm not sure this is a pattern we would like to encourage outside of PyTorch core abstractions.
Thoughts?

Reply:
Agreed. Apart from the availability, this is also hard to discover. Unless you grep for `decode_sample`, you probably won't find the `decode_sample` datapipe.

Reply:
So let's remove this from torchvision and be more verbose on our end, wdyt?

Reply:
See https://github.com/pytorch/vision/pull/4441#issuecomment-921851202